src/main/java/com/spring/app/services/mappers/CountryProfileDTOResultMapper.java

```diff
@@ -0,0 +1,67 @@
+package com.spring.app.services.mappers;
+
+import com.spring.app.DataNotFoundException;
+import com.spring.app.EconomicLevel;
+import com.spring.app.dto.CountryProfileDTO;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Maps data set to pairs of country profiles (as a value) and countries name (as a key)
+ *
+ * @author Lyubov Ruzanova
+ */
+public class CountryProfileDTOResultMapper implements ResultMapper<List<Object[]>, Map<String, CountryProfileDTO>> {
+
+    private static final int COUNTRY_INDEX = 0;
+    private static final int ECONOMY_INDEX = 1;
+    private static final int VAC_NUMBER_INDEX = 2;
+    private static final int EMP_NUMBER_INDEX = 3;
+
+    /**
+     * {@InheritDoc}
+     */
+    @Override
+    public Map<String, CountryProfileDTO> map(List<Object[]> objects) {
+        if (objects != null && !objects.isEmpty()) {
+            Map<String, CountryProfileDTO> dtoHashMap = new HashMap<>();
+            for (Object[] object : objects) {
+                if (object != null) {
+                    String name = resolveStringValue(object[COUNTRY_INDEX]);
+                    EconomicLevel economyIndex = EconomyLevelResultMapper.getEconomicLevel(object[ECONOMY_INDEX]);
+                    Long vacNumber = LongResultMapper.getLong(object[VAC_NUMBER_INDEX]);
+                    Long empNumber = LongResultMapper.getLong(object[EMP_NUMBER_INDEX]);
+
+                    CountryProfileDTO countryProfileDTO = new CountryProfileDTO();
+                    countryProfileDTO.setNumberOfEmployees(empNumber);
+                    countryProfileDTO.setNumberOfVacancies(vacNumber);
+                    countryProfileDTO.setEconomicLevel(economyIndex);
+
+                    dtoHashMap.put(name, countryProfileDTO);
+                } else {
+                    throw new DataNotFoundException("The received value is incorrect.");
```

Comment:
Message in constants, do not throw any exceptions in mapper. Just return an empty collection

Reply:
Oh, really) You are right :) You mean there is no mistake, it's just no-result response from DB. I missed it. But if I find out a that result isn't instance of object that I want I will throw an error?